rio_tiler/mosaic/reader.py

```diff
@@ -80,7 +83,7 @@ def reader(asset: str, *args, **kwargs) -> Tuple[numpy.ndarray, numpy.ndarray]:
     for chunks in _chunks(assets, chunk_size):
         tasks = create_tasks(reader, chunks, threads, *args, **kwargs)
         for (t, m), asset in filter_tasks(
-            tasks, allowed_exceptions=(TileOutsideBounds,)
+            tasks, allowed_exceptions=allowed_exceptions,
```

Comment:
A good question is whether we should always include `TileOutsideBounds`. A user who doesn't know the internals of the API might provide `allowed_exceptions=(botocore.NoSuchKey)` and then be confused why tons of new errors are coming up from `TileOutsideBounds`.

E.g. should `allowed_exceptions` be _added_ to a default of `TileOutsideBounds`?

Reply:
I think the most intuitive default behavior is to reraise any exception not in `allowed_exceptions`, having a default set of exceptions that are always makes this more confusing.

Reply:
That's fair. Probably good to add this to the docstring that if you provide a custom list you need to include `TileOutsideBounds` explicitly if desired.